### DISCUSSION TITLE
Operators

### DISCUSSION THREAD

**Discussion by lukaszgrudnik** [ReactiveX/rxjs#7491] [Ideas / Feature request]
I encountered an issue that I needed to subscribe several roughly 11 different observables. This problem can be solved by using zip operator, but this operator returns array that is hard do cope with in code. It's not as readable as I would wish to.Maybe you encountered such problem as well, what do you think about custom operator like this in the exmaple below ?

interface UserData {...}

interface CarData {...}

interface Address {...}

zipLikeOperator(
    {
        users: Observable<UserData[]>, 
        cars: Observable<CarData[]>,
        addresses: Observable<Address[]>
    }
).subscribe((data: {users: UserData[], cars: CarData[], adressess: Address[]})=> {
    
})

**Comment by jakovljevic-mladen** [comment]
You can use [`map`](https://rxjs.dev/api/index/function/map) operator to transform array to an object:

```typescript
zip(users, cars, addresses).pipe(
  map(([users, cars, addresses]) => ({ users, cars, addresses }))
)
```

**Reply by lukaszgrudnik** [reply]
Quite straightforward but less elegant when you have many fields
